sylvia-derive/src/interface/communication/api.rs

```diff
@@ -58,45 +49,27 @@ impl<'a> Api<'a> {
             &source.generics.where_clause,
         );
 
-        let where_clause = &self.associated_types.as_where_clause();
-        let custom_query = custom.query_or_default();
         let exec_generics = &exec_variants.used_generics();
         let query_generics = &query_variants.used_generics();
         let sudo_generics = &sudo_variants.used_generics();
 
-        let phantom = if !generics.is_empty() {
-            quote! {
-                _phantom: std::marker::PhantomData<( #(#generics,)* )>,
-            }
-        } else {
-            quote! {}
-        };
-
         quote! {
             pub trait InterfaceMessagesApi {
                 type Exec;
                 type Query;
                 type Sudo;
-                type Querier<'querier>;
             }
 
             impl<Contract: #interface_name> InterfaceMessagesApi for Contract {
                 type Exec = ExecMsg < #(<Contract as #interface_name >:: #exec_generics,)* >;
                 type Query = QueryMsg < #(<Contract as #interface_name >:: #query_generics,)* >;
                 type Sudo = SudoMsg < #(<Contract as #interface_name >:: #sudo_generics ,)* >;
-                type Querier<'querier> = #sylvia ::types::BoundQuerier<'querier, #custom_query, Contract >;
-            }
-
-
-            pub struct Api < #(#generics,)* > {
-                #phantom
             }
 
-            impl < #(#generics,)* > #sylvia ::types::InterfaceApi for Api < #(#generics,)* > #where_clause {
+            impl<'a, Error, #(#generics),*> InterfaceMessagesApi for dyn #interface_name < Error = Error, #(#generics = #generics,)* > + 'a {
```

Comment:
It should not be an issue here, but please keep the convention of naming generated lifetimes and make it more unique with `sv_` prefix

Reply:
Done